### ISSUE TITLE
A match_kind to match only identifiers at callsites?

### ISSUE THREAD

**Issue by brennantaylor** [realm/SwiftLint#1961]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### match_kind Request

We have some use-cases to blacklist some method calls or mutating of properties.

However identifier matches parameters and arguments as well giving us false positives. Was wondering if there is a chance to get a match_kind that only matches callsites.

For example we want to blacklist and call to `sleep(Int)` in our code but not restrict people from having the parameter name.

We've worked around it by just writing a regex for ` sleep\\(` but given how we've used match_kinds to simplify some other custom rules we have our team was wondering if something to handle this makes sense or what it would even look like.

Or if we're misusing match_kinds please let us know.

**Comment by marcelofabri** [comment]
@brennantaylor Syntax kinds only tell you about syntax: "is this token a string literal, an identifier, a keyword, etc?"

A simple function call token like `sleep` has the kind of an `identifier`. You can check this in the future by running `https://github.com/jpsim/SourceKitten`.

For example:

```
sourcekitten syntax --text "sleep(10)"
[
  {
    "type" : "source.lang.swift.syntaxtype.identifier",
    "offset" : 0,
    "length" : 5
  },
  {
    "type" : "source.lang.swift.syntaxtype.number",
    "offset" : 6,
    "length" : 2
  }
]
```

I'm closing this for now. Feel free to reopen if you have more questions.
